File: stream/src/test/java/bsky4j/stream/atproto/sync/SubscribeRepoTest.java
package kbsky.stream.atproto.sync;

import kbsky.domain.Service;
import kbsky.stream.ATProtocolStreamFactory;
import kbsky.stream.AbstractTest;
import kbsky.stream.api.entity.atproto.sync.SyncSubscribeReposRequest;
import kbsky.stream.util.StreamClient;
import net.socialhub.logger.Logger;
import org.junit.Test;

public class SubscribeRepoTest extends AbstractTest {

    @Test
    public void testSubscribeRepo() {
        Logger.getLogger(null).setLogLevel(Logger.LogLevel.WARN);

        try {
            StreamClient stream = ATProtocolStreamFactory
                    .getInstance(Service.BSKY_SOCIAL.getUri())
                    .sync().subscribeRepos(
                            SyncSubscribeReposRequest
                                    .builder()
                                    .build()
                    );

            stream.setEventCallback((cid, url, record) -> {
                print(url, record);
            });

            stream.connect();
            Thread.sleep(10000);

        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: stream/src/jvmTest/kotlin/work/socialhub/kbsky/stream/atproto/sync/SubscribeRepoTest.kt
package work.socialhub.kbsky.stream.atproto.sync

import kotlinx.coroutines.delay
import kotlinx.coroutines.launch
import kotlinx.coroutines.runBlocking
import work.socialhub.kbsky.model.share.RecordUnion
import work.socialhub.kbsky.stream.ATProtocolStreamFactory
import work.socialhub.kbsky.stream.AbstractTest
import work.socialhub.kbsky.stream.api.entity.com.atproto.SyncSubscribeReposRequest
import work.socialhub.kbsky.stream.entity.callback.OpenedCallback
import work.socialhub.kbsky.stream.entity.com.atproto.callback.SyncEventCallback
import kotlin.test.Test

class SubscribeRepoTest : AbstractTest() {

    @Test
    fun testSubscribeRepo() {
        runBlocking {
            val stream = ATProtocolStreamFactory
                .instance()
                .sync()
                .subscribeRepos(
                    SyncSubscribeReposRequest().also {
                        it.filter = listOf(
                            "app.bsky.feed.post"
                        )
                    }
                )

            stream.eventCallback(
                object : SyncEventCallback {
                    override fun onEvent(
                        cid: String?,
                        uri: String?,
                        record: RecordUnion
                    ) {
                        print(record)
                    }
                })

            stream.openedCallback(
                object : OpenedCallback {
                    override fun onOpened() {
                        println(">> opened.")
                    }
                })


            launch { stream.open() }.let {
                delay(10000)
                it.cancel()
                stream.close()
            }
        }
    }
}
